src/battle-scene.ts

```diff
@@ -2064,6 +2064,29 @@ export default class BattleScene extends SceneBase {
     }
   }
 
+  /**
+   * Tries to add the input phase to right before the MoveEndPhase, useful for resolving moves before the end-of-turn phases start
+   * @param phase {@linkcode Phase} the phase to be added
+   * @returns boolean if a MoveEndPhase was found
+   */
+  prependToMoveEndPhase(phase: Phase): boolean {
+    let flag = false;
+    // want to add the pahse after MoveEnd of U-turn? so it can ge the effects of whatever
+    for (let i = 0; i < this.phaseQueue.length; i++) {
+      if (this.phaseQueue[i].constructor.name === "MoveEndPhase") {
+        this.phaseQueue.splice(i, 0, phase);
+        flag = true;
+        break;
+      }
+    }
+    if (flag === false) {
+      // what to do here if there is none?
+      // it should be the case that during combat there is always a MoveEndPhase when a move is being used, extensive testing/reasoning needed
+      this.unshiftPhase(phase);
+    }
+    return flag;
+  }
```

Comment:
```suggestion
    const targetIndex = this.phaseQueue.findIndex(ph => ph instanceof targetPhase);
    if (targetIndex !== -1) {
        this.phaseQueue.splice(targetIndex, 0, phaseToAdd);
        return true;
    } else {
        this.unshiftPhase(phaseToAdd);
        return false;
    }
  }
```

To go along with the change to the function definition (also using `findIndex` which performs better than a for loop).